imagecontent/storage.conf

```diff
@@ -27,10 +28,7 @@ size = ""
 
 # Path to an helper program to use for mounting the file system instead of mounting it
 # directly.
-#mount_program = "/usr/bin/fuse-overlayfs"
-
-# OverrideKernelCheck tells the driver to ignore kernel checks based on kernel version
-override_kernel_check = "false"
+mount_program = "/usr/bin/fuse-overlayfs"
```

Comment:
This forces us to use fuse-overlayfs even when we're running with privileges.

Reply:
I'd prefer that we wire in a variable somewhere around [here](https://github.com/openshift/builder/blob/9277419505b020ca50c35ce1c045526413249c93/pkg/build/builder/cmd/builder.go#L118), that the build controller can set, that would let it tell us to use this when it's running us without privileges.

Reply:
We could do this, but fuse-overlay will work with --privilege as well.

Reply:
Added a check for BUILD_PRIVILEGED, and drop any GraphDriverOptions if set.

Reply:
That would require the build controller to start setting the variable to continue to get the current behavior, which I'm not really in favor of, since updating the builder image before updating the controller would break builds.  A method for passing in driver options that the controller can use to set a `mount_program` would work, without imposing that ordering/upgrade requirement.

Reply:
@nalind Now I pass in a "," list of options in the BUILD_STORAGE_OPTIONS environment variable?  Is this what you want?

Reply:
Options like "mountopt" include commas, so in the interest of not having to come up with an escaping scheme or having to change this later, I'd suggest expecting it to contain a JSON-encoded slice of strings that we unmarshal and use directly.  The value is set by the controller logic when it creates the pod spec that ultimately controls how a container is started using the image, and the controller already sets JSON-encoded values in environment variables for more complicated structures.